src/core/ext/client_channel/client_channel.c

```diff
@@ -546,10 +546,14 @@ static grpc_error *cc_init_channel_elem(grpc_exec_ctx *exec_ctx,
   arg = grpc_channel_args_find(args->channel_args, GRPC_ARG_SERVER_URI);
   GPR_ASSERT(arg != NULL);
   GPR_ASSERT(arg->type == GRPC_ARG_STRING);
-  chand->server_name = gpr_strdup(arg->value.string);
+  grpc_uri *uri = grpc_uri_parse(arg->value.string, true);
+  GPR_ASSERT(uri->path[0] != '\0');
```

Comment:
Ah, https://github.com/grpc/grpc/pull/9539/commits/4d1bc51b5acc549157c129c82e9fcd109aad2b2c, precisely.

The comment about checking eagerly still stands.